score-server/src/test/java/bio/overture/score/server/security/JWTSecurityTest.java

```diff
@@ -174,33 +204,33 @@ public void validateCanUploadWithValidAuthScopes() {
         return ResponseEntity.status(mvcResponse.getStatus()).body(responseObject);
     }
 
+    private final Map<ScopeOptions, List<String>> downloadScopesMap = Map.of(
+            VALID_SYSTEM,   List.of(resolveSystemDownloadScope(), "score.READ", "id.READ"),
+            VALID_STUDY,    List.of(resolveStudyDownloadScope(EXISTING_PROJECT_ID), "id.READ"),
+            INVALID_STUDY,  List.of(resolveStudyDownloadScope(NON_EXISTING_PROJECT_ID), "id.READ")
+    );
+
+    private final Map<ScopeOptions, List<String>> uploadScopesMap = Map.of(
+            VALID_SYSTEM, List.of(resolveSystemUploadScope(), "score.WRITE", "id.WRITE"),
+            VALID_STUDY, List.of(resolveStudyUploadScope(EXISTING_PROJECT_ID), "id.WRITE"),
+            INVALID_STUDY, List.of(resolveStudyUploadScope(NON_EXISTING_PROJECT_ID), "id.WRITE")
+    );
```

Comment:
in general, this is a risky way to define a final member variables. those resolve methods depend on non-final member variable (securityConfig). Since final member variables can only be defined at construction/initialization time, that means the non-final member `secruityConfig` must be defined before the final variable is defined. I believe this coincidentally works, because of spring magic, however if there was no spring magic, `securityConfig` would be null by the time the final variables are defined, resulting in NullPointerExceptions.

The following is an example of this:
```java
  @AllArgsConstructor
  @NoArgsConstructor
  public class Car {

    private String name;

    private final String something = getNewName();

    public String getSomething(){
      return something;
    }

    public String getNewName(){
      return name+System.currentTimeMillis();
    }
  }

```

the following Test passes
```java
    val c = new Car();
    assertEquals("null_test123",c.getSomething());

    val c2 = new Car("john");
    assertEquals("john", c2.getName());
    assertNotEquals("john_test123", c2.getSomething());
    assertEquals("null_test123", c2.getSomething());

```

One solution is to define this in the `beforeEachTest` method. You could rewrite it as follows:
```java
    private Map<ScopeOptions, List<String>> uploadScopesMap;
    private Map<ScopeOptions, List<String>> downloadScopesMap;

    @Before    
    @SneakyThrows    
    public void beforeEachTest() {
        jwtGenerator = new JWTGenerator(keyPair);
        val notInitialized = this.mockMvc == null
        if (notInitialized) {
            this.mockMvc =
                    MockMvcBuilders.webAppContextSetup(webApplicationContext).apply(springSecurity()).build();

            this.uploadScopesMap = Map.of(
            VALID_SYSTEM, List.of(resolveSystemUploadScope(), "id.WRITE"),
            VALID_STUDY, List.of(resolveStudyUploadScope(EXISTING_PROJECT_ID), "id.WRITE"),
            INVALID_STUDY, List.of(resolveStudyUploadScope(NON_EXISTING_PROJECT_ID), "id.WRITE"));

            this.downloadScopesMap = Map.of(
            VALID_SYSTEM,   List.of(resolveSystemDownloadScope(), "id.READ"),
            VALID_STUDY,    List.of(resolveStudyDownloadScope(EXISTING_PROJECT_ID), "id.READ"),
            INVALID_STUDY,  List.of(resolveStudyDownloadScope(NON_EXISTING_PROJECT_ID), "id.READ"));
        }
        setMocks();

    }
```

Reply:
Very good point, spring magic has spoiled me :stuck_out_tongue:. Will fix :+1: